core/src/main/java/org/aerogear/mobile/core/metrics/MetricsService.java

```diff
@@ -71,31 +67,35 @@ public void sendAppAndDeviceMetrics() {
      * @param callback callback of the publication
      */
     public void sendAppAndDeviceMetrics(final Callback callback) {
-        this.publish(defaultMetrics, callback);
+        this.publish(INIT_METRICS_TYPE, new Metrics[0], callback);
     }
 
     /**
      * Send metrics
      *
+     * @param type type of the enclosing metrics event
      * @param metrics Metrics to send
      */
-    public void publish(Metrics... metrics) {
-        publish(metrics, null);
+    public void publish(String type, Metrics... metrics) {
+        publish(type, metrics, null);
     }
 
     /**
      * Send metrics
      *
+     * @param type type of the enclosing metrics event
      * @param metrics Metrics to send
      * @param callback callback of the publication
      */
-    public void publish(@NonNull final Metrics[] metrics, final Callback callback) {
+    public void publish(@NonNull String type, @NonNull final Metrics[] metrics,
+                    final Callback callback) {
         if (publisher == null) {
             throw new IllegalStateException(
                             "Make sure you have called configure or get this instance from MobileCore.getInstance()");
         }
+        nonNull(type, "type");
         nonNull(metrics, "metrics");
-        publisher.publish(metrics, callback);
+        publisher.publish(type, metrics, callback);
```

Comment:
`nonNull` returns the passed in object, so this could be changed to a one liner

Reply:
will do

Reply:
@ziccardi 
```
publisher.publish(nonNull(type, "type"), nonNull(metrics, "metrics"), callback);
```
This looks ugly IMO, when there are multiple parameters.
I will change it only here. Won't change it for all other occurrences in the code.

Reply:
@aliok : yeah, you are right :D probably would look better formatting the code as:
```java
publisher.publish(
              nonNull(type, "type"), 
              nonNull(metrics, "metrics"), 
              callback);
```

but making the checks alone at the start of the method would be probably better

Reply:
```
publisher.publish(
              nonNull(type, "type"), 
              nonNull(metrics, "metrics"), 
              callback);
```
I went with that but our linter doesn't like that :)